solr/core/src/java/org/apache/solr/core/SchemaCodecFactory.java

```diff
@@ -124,18 +124,18 @@ public KnnVectorsFormat getKnnVectorsFormatForField(String field) {
             if (fieldType instanceof DenseVectorField) {
               DenseVectorField vectorType = (DenseVectorField) fieldType;
               String knnAlgorithm = vectorType.getKnnAlgorithm();
-              if (knnAlgorithm != null) {
-                if (knnAlgorithm.equals(DenseVectorField.HNSW_ALGORITHM)) {
-                  int maxConn = vectorType.getHnswMaxConn();
-                  int beamWidth = vectorType.getHnswBeamWidth();
-                  return new Lucene94HnswVectorsFormat(maxConn, beamWidth);
-                }
+              if (knnAlgorithm.equals(DenseVectorField.HNSW_ALGORITHM)) {
```

Comment:
nit the following will avoid any NPE if `knnAlgorithm` is ever null.: 
```suggestion
              if (DenseVectorField.HNSW_ALGORITHM.equals(knnAlgorithm)) {
```